src/careamics/careamist.py

```diff
@@ -594,25 +598,59 @@ def predict(
         -------
         list of NDArray or NDArray
             Predictions made by the model.
+
+        Raises
+        ------
+        ValueError
+            If mean and std are not provided in the configuration.
+        ValueError
+            If tile size is not divisible by 2**depth for UNet models.
+        ValueError
+            If tile overlap is not specified.
         """
-        # create inference configuration using the main config
-        inference_dict: dict = create_inference_parameters(
-            configuration=self.cfg,
-            tile_size=tile_size,
-            tile_overlap=tile_overlap,
-            data_type=data_type,
-            axes=axes,
-            tta_transforms=tta_transforms,
-            batch_size=batch_size,
-        )
+        if (
+            self.cfg.data_config.image_means is None
+            or self.cfg.data_config.image_stds is None
+        ):
+            raise ValueError("Mean and std must be provided in the configuration.")
+
+        # tile size for UNets
+        if tile_size is not None:
+            model = self.cfg.algorithm_config.model
+
+            if model.architecture == SupportedArchitecture.UNET.value:
+                # tile size must be equal to k*2^n, where n is the number of pooling
+                # layers (equal to the depth) and k is an integer
```

Comment:
Ah, I realise now the `InferenceConfig` doesn't have access to the type of architecture so it can't be validated in there 😅, I guess it might be nice to isolate this check as a separate function somewhere but otherwise looks good!

Reply:
Not only the type of architecture, but also the depth of the UNet...

But you had a good point with the double `Inference` instantiation, that was convoluted. Let's see later if we want to move that to some utils package or so.